docs/fxa-engineering/fxa-tests-circleci.md

```diff
@@ -0,0 +1,278 @@
+
+For most modules: if `Dockerfile-test` is present, it's used to build a Docker container.
+
+The container is tagged `${MODULE}:test`, where `MODULE` is named in the first parameter of the script. Then, `npm run test` is run within the container - the name of the npm script can be customized with the second parameter of `test.sh`.
+
+If `eslint` or `tslint` is present in the module's `package.json`, then `npm run lint` is also run in the container. If there's a `Gruntfile.js` that contains `eslint`, then `grunt eslint` will be run in the container.
+
+### .circleci/deploy.sh
+
+Common script for deploying the Docker container built from an FxA package to Docker Hub.
+
+The basic gist of this script is that it takes a `{MODULE}:build` container - e.g. as created via `build.sh` - and re-tags it as appropriate for the current CI run before pushing it to Docker Hub.
+
+These images are deployed to Mozilla's Docker Hub account. So, for example, many modules can be found with a search for [`mozilla/fxa-`](https://hub.docker.com/search?q=mozilla%2Ffxa-&type=image). Also look for images such as [`mozilla/123done`](https://hub.docker.com/r/mozilla/123done) and [`mozilla/browserid-verifier`](https://hub.docker.com/r/mozilla/browserid-verifier), which are listed under `packages` but do not follow the `fxa-*` naming convention.
+
+The rest of the tag name depends on a few other conditions:
+
+If the build is on `master` branch, then the tag will be `{MODULE}:latest`.
+
+If a build is tagged in Git, then that Git tag is used - i.e. `{MODULE}:{GIT_TAG_NAME}`.
+
+If a build is on a branch with a name prefixed with `feature` or `dockerpush`, that branch name will be used in the tag - i.e. `{MODULE}:{GIT_BRANCH_NAME}`.
+
+It's also possible to supply `$MODULE_SUFFIX` env var tweak the Docker Hub repo name - i.e. `mozilla/$MODULE-$MODULE_SUFFIX`.
+
+Untagged commits or commits on branches that do fall into one of the above cases do not result in a container deployed to Docker Hub.
+
+Logic for running or skipping deployment for a given module is also implemented here: `packages/test.list` is expected to contain a list output from `modules-to-test.js` - if the current module's name is missing and the list isn't `all`, this script will exit immediately.
+
+### .circleci/tag.sh
+
+Used solely by `fxa-email-service` as a custom version of `build.sh` and `deploy.sh`:
+
+- Builds a Docker container using `Dockerfile-tag` tagged `${MODULE}:latest`
+- Re-tags the Docker container using basically the same logic as `deploy.sh` and pushes to Docker Hub
+
+### .circleci/install-content-server.sh
+
+Copies over `version.json` and runs `npm install` for fxa-content-server.
+
+### .circleci/test-content-server.sh
+
+TBD:
```

Comment:
And this is where I can kind of come in. @vladikoff can also fill in some holes in my knowledge.